sdk/dotnet/Pulumi.Tests/Mocks/MocksTests.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Pulumi.Testing;
+using Xunit;
+
+namespace Pulumi.Tests.Mocks
+{
+    class MyMocks : IMocks
+    {
+        public Task<object> CallAsync(string token, ImmutableDictionary<string, object> args, string? provider)
+        {
+            return Task.FromResult<object>(args);
+        }
+
+        public Task<(string id, object state)> NewResourceAsync(string type, string name, ImmutableDictionary<string, object> inputs, string? provider, string? id)
+        {
+            Assert.NotEqual("pulumi:pulumi:Stack", type);
+
+            switch (type)
+            {
+                case "aws:ec2/instance:Instance":
+                    return Task.FromResult<(string, object)>(("i-1234567890abcdef0", new Dictionary<string, object> {
+                        { "publicIp", "203.0.113.12" },
+                    }));
+
+                default:
+                    return Task.FromResult<(string, object)>(("", new Dictionary<string, object>()));
```

Comment:
Should we fail by default if anything except the Instance resource is newed-up? That seems like a stronger test.